woocommerce/woocommerce
# Add new payment method icons component

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




Closes [#8808](https://github.com/Automattic/woocommerce-payments/issues/8808) .

Create a new component that can be reused, to display the new payment method logos.
Design: https://www.figma.com/design/VLWDmeuIXQ7XAK4flyqTFO/Payments-Adoption-%26-Activation?node-id=2196-11373&t=7gXadrjGDmPqgpUP-0


### How to test the changes in this Pull Request:



Environment setup: paJDYF-cHI-p2 

The new component is implemented in:
1. Payments Settings Page
-  WooPayments should not be installed.
<img width="1715" alt="Screenshot 2024-06-21 at 08 11 43" src="https://github.com/woocommerce/woocommerce/assets/82826872/0ebf59d4-cd98-47ff-97e6-30e76f6a1f20">

2. Woocommerce home page - Connect Page 
- WooPayments is not installed ; 
- no incentives can be applied 
- URL: http://localhost:8082/wp-admin/admin.php?page=wc-admin&task=payments

<img width="712" alt="Screenshot 2024-06-21 at 11 35 14" src="https://github.com/woocommerce/woocommerce/assets/82826872/3a01749c-c937-428c-a4d7-5f594ee0784d">

3. Woocommerce connect page with incentive 
- URL: http://localhost:8082/wp-admin/admin.php?page=wc-admin&path=%2Fwc-pay-welcome-page&from=WCADMIN_PAYMENT_TASK 
- WooPayments should not be installed
- In order to activate incentives, you have to change [this](https://github.com/woocommerce/woocommerce/blob/add/8808-woo-payment-method-logo/plugins/woocommerce/src/Internal/Admin/WcPayWelcomePage.php#L382)  with the following:


```
		$store_context = [
			// Store ISO-2 country code, e.g. `US`.
			'country'      => WC()->countries->get_base_country(),
			// Store locale, e.g. `en_US`.
			'locale'       => get_locale(),
			// WooCommerce store active for duration in seconds.
			'active_for'   => 100,
			'has_orders'   => $this->has_orders(),
			// Whether the store has at least one payment gateway enabled.
//			'has_payments' => ! empty( WC()->payment_gateways()->get_available_payment_gateways() ),
//			'has_wcpay'    => $this->has_wcpay(),
		];
```
<img width="729" alt="Screenshot 2024-06-21 at 12 43 10" src="https://github.com/woocommerce/woocommerce/assets/82826872/0df14999-cb89-4b06-a7e9-abb1e95e73ab">
 
4. Payments Settings Page - when WooPayments is installed
- WooPayments IS installed
- after installing WooPayments go to the Payments settings page and disable WooPayments

<img width="1712" alt="Screenshot 2024-06-21 at 13 01 28" src="https://github.com/woocommerce/woocommerce/assets/82826872/c5a1792c-b5d3-46b0-b6ae-45cd3f08ee77">



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Add new payment method icons component
</details>
